zerver/migrations/0468_backfill_edit_history_json.py

```diff
@@ -0,0 +1,46 @@
+# Generated by Django 3.2.2 on 2021-05-11 10:00
+import orjson
+from django.db import migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+
+BATCH_SIZE = 5000
+
+
+def backfill_edit_messages(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    Message = apps.get_model("zerver", "Message")
+    ArchivedMessage = apps.get_model("zerver", "ArchivedMessage")
+    msg_models = [Message, ArchivedMessage]
+
+    for model in msg_models:
+        if not model.objects.filter(edit_history__isnull=False).exists():
+            return
+
+        messages = model.objects.filter(edit_history__isnull=False)
+        id_lower_bound = messages.earliest("id").id
+        id_upper_bound = messages.latest("id").id
+
+        while id_lower_bound <= id_upper_bound:
+            current_id_upper_bound = min(id_lower_bound + BATCH_SIZE, id_upper_bound)
+            with transaction.atomic():
+                messages = model.objects.filter(
+                    edit_history__isnull=False, id__range=(id_lower_bound, current_id_upper_bound)
+                )
+                for message in messages:
+                    message.edit_history_json = orjson.loads(message.edit_history)
+                model.objects.bulk_update(messages, ["edit_history_json"])
```

Comment:
Let's pass `batch_size=250` to this `bulk_update` call to limit the scope of quadratic behavior. Because edited messages are sparse, it'll often be less than 250 anyway.

Also, since this is a Message table migration and likely quite slow, we want a progress indicator; a bunch of our older Message migrations have them. `zerver/migrations/0419_backfill_message_realm.py`.